### PR TITLE
Fix storage of attributes revision

### PR DESCRIPTION
**Pull request by coolo**
I'll rebase on #5165 once merged - fixing this requires quite some cassette updates, so I would rather have this as extra

### CODE REVIEW

File: src/api/spec/factories/attribs.rb [hunk 1]
```diff
@@ -7,11 +7,16 @@
...
+    before(:create) do |attrib, _evaluator|
+      User.current = User.find_by_login!(attrib.creator)
+    end
+
     after(:create) do |attrib, evaluator|
       if evaluator.package
         attrib.package = evaluator.package
         attrib.project = nil
       end
+      User.current = nil
```
Comment (hennevogel) [inline src/api/spec/factories/attribs.rb]: Why shouldn't this be valid?

```ruby
RSpec.describe BLAH, type: :model do
  let(:attribute) { create(:attrib) }
  let(:user] { create(:confirmed_user) }

  describe '#whatever' do
    login(user)
    it { expect(attribute.fullname).to eq("#{attribute.namespace}:#{attribute.name}") }
  end
end
```

Comment (coolo) [reply src/api/spec/factories/attribs.rb]: the problem is that the create(:attrib) in your 2nd line will already crash. I changed this now to build

File: src/api/spec/features/webui/attributes_spec.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 RSpec.feature 'Attributes', type: :feature, js: true do
   let!(:user) { create(:confirmed_user) }
   let!(:attribute_type) { create(:attrib_type) }
-  let!(:attribute) { create(:attrib, project_id: user.home_project.id) }
+  let!(:attribute) { create(:attrib, project_id: user.home_project.id, creator: user) }
```
[the remark this answers is not in the document]
Comment (coolo) [reply src/api/spec/features/webui/attributes_spec.rb]: For the same reason the factory of bs_request is setting User.current I assume

File: src/api/spec/factories/attribs.rb (hunk 1 above)
Comment (hennevogel) [reply src/api/spec/factories/attribs.rb]: How would it crash if User.current is set?

File: src/api/spec/features/webui/attributes_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (hennevogel) [reply src/api/spec/features/webui/attributes_spec.rb]: And that is?

File: src/api/spec/factories/attribs.rb (hunk 1 above)
Comment (coolo) [reply src/api/spec/factories/attribs.rb]: I didn't spot the difference between let and let!

Comment (hennevogel) [reply src/api/spec/factories/attribs.rb]: Happened to all of us

**Comment by coolo** [comment]
There seems to be a generic problem with factorybot and vcr - e.g. when I recreate the cassettes I get 
diffs like:

```
--- a/src/api/spec/cassettes/Attributes/for_a_project_with_a_package/add_attribute_with_values.yml
+++ b/src/api/spec/cassettes/Attributes/for_a_project_with_a_package/add_attribute_with_values.yml
@@ -2,1065 +2,12 @@
 http_interactions:
 - request:
     method: put
-    uri: http://backend:5352/source/home:user_10/_project/_attribute?meta=1&user=user_9
+    uri: http://backend:5352/source/home:user_210/package_109/_attribute?meta=1&user=user_210

**Comment by bgeuken** [comment]
@coolo If your test relies on the backend / VCR you need to keep the requested URI static, e.g. if your request includes the package name you should create the package factory like this: ` create(:package, name: 'foo')`. Otherwise the URI changes with each test run.

Though your diff compares different endpoints. So this might be caused by something else.

**Comment by coolo** [comment]
I figured if I skip the backend put based on global_write_through I don't have to adapt half as many tests

**Comment by coolo** [comment]
merged back into 5165 as it was reviewed and passes tests

**Pull request closed without merging**